src/main/java/picard/sam/markduplicates/UmiGraph.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package picard.sam.markduplicates;
+
+import htsjdk.samtools.DuplicateSet;
+import htsjdk.samtools.SAMRecord;
+import picard.PicardException;
+
+import java.util.*;
+
+public class UmiGraph {
+    private Map<String, Long> umiCounts;
+    private int[] duplicateSet;    // id[i] = parent of i
+    private Long[] occupancy;
+    private String[] umi;
+    private int editDistanceToJoin;
+    private int nUmis;
+    private int nDuplicateSets;
+
+    UmiGraph() {
+
+    }
+
+    UmiGraph(final Map<String, Long> umiCounts, final int editDistanceToJoin) {
```

Comment:
Should this class also be package protected, or should this be public?  I could see this class being useful.